src/plugin.js

```diff
@@ -740,6 +743,35 @@ const contribAdsPlugin = function(options) {
 
   };
 
+  /**
+   * iOS Safari will change caption mode to 'showing' if a user previously
+   * turned captions on manually for that video source, so this TextTrackList
+   * 'change' event handler will re-disable them in case that occurs during ad playback
+   */
+  const iOSTrackListChangeHandler = function() {
+    const textTrackList = player.textTracks();
+
+    // Exit early if the platform does not match or this occurs during content
+    // If shouldPlayContentBehindAd, no special handling is needed
+    if (player.ads.shouldPlayContentBehindAd(player) || !player.ads.isAdPlaying() ||
+      !player.tech_.featuresNativeTextTracks || !videojs.browser.IS_IOS ||
+      Array.isArray(textTrackList)) {
+      return;
+    }
+
+    for (let i = 0; i < textTrackList.length; i++) {
+      const track = textTrackList[i];
+
+      if (track.mode === 'showing') {
+        track.mode = 'disabled';
+      }
+    }
+  };
+
+  player.ads._trackChangeDuringAdsHandler = iOSTrackListChangeHandler;
```

Comment:
We don't need to store this on the ads object anymore.

Reply:
Don't we need it to remove the listener on `dispose` still?

Reply:
We do, and we can remove `iOSTrackListChangeHandler` rather than a reference stored on the `ads` object. It's the same value.

Reply:
Ah, yes `iOSTrackListChangeHandler` is in scope here, my mistake!